codegenerator/cli/templates/static/codegen/src/eventFetching/ChainManager.res

```diff
@@ -72,12 +72,35 @@ type fetchStateWithData = {
   heighestBlockBelowThreshold: int,
 }
 
+let isQueueItemEarlierUnorderedBelowReorgThreshold = (
+  ~fetchStatesMap: ChainMap.t<fetchStateWithData>,
+) => (a: multiChainEventComparitor, b: multiChainEventComparitor) => {
+  let isItemBelowReorgThreshold = item => {
+    item.earliestEvent->FetchState.queueItemIsInReorgThreshold(
+      ~heighestBlockBelowThreshold=ChainMap.get(
+        fetchStatesMap,
+        a.chain,
+      ).heighestBlockBelowThreshold,
+    )
+  }
+  switch (a->isItemBelowReorgThreshold, b->isItemBelowReorgThreshold) {
+  | (false, true) => true
+  | (true, false) => false
+  | _ => isQueueItemEarlierUnorderedMultichain(a, b)
+  }
+}
+
```

Comment:
The idea here is if we are in undordered multichain mode, always prioritize queue items that are below the reorg threshold. That way we can register contracts all the way up to the threshold on all chains before starting. Similarly we wait till all chains are at their threshold before saving entity history.

Reply:
Could you leave it as a comment in the code.